### ISSUE TITLE
EFS with infrequent access life cycle

### ISSUE THREAD

**Issue by mnikhil-git** [terraform-providers/terraform-provider-aws#7623]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

EFS now allows for setting life cycle / IA configuration for saving costs on infrequently accessed data. This seems to be for new EFS filesystems only though.

Please have terraform efs resource to leverage and enable this configuration
https://aws.amazon.com/about-aws/whats-new/2019/02/amazon-efs-introduces-lower-cost-storage-class/

### New or Affected Resource(s)

* aws_efs_*

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by bflad** [comment]
It appears this should be implemented via a new `lifecycle_policy` configuration block in `aws_efs_file_system`, e.g.

```hcl
# Potential Terraform configuration, details may change during development
resource "aws_efs_file_system" "example" {
  # ... other configuration ...
  lifecycle_policy {
    transition_to_ia = "AFTER_30_DAYS"
  }
}
```

**Comment by mnikhil-git** [comment]
+1

**Comment by mnikhil-git** [comment]
this would be an AWS specific question I guess. In the event that terraform resource is updated with the new lifecycle policy change, can I make changes to an existing EFS filesystem using this change in the resource block?

**Comment by mnikhil-git** [comment]
hi @bflad  just checking for your suggestions on this

**Comment by tdmalone** [comment]
> can I make changes to an existing EFS filesystem using this change in the resource block?
According to [the docs](https://docs.aws.amazon.com/efs/latest/ug/lifecycle-management-efs.html), only if you created your file system after 13th Feb this year.

**Comment by ryno75** [comment]
@bflad anybody working this issue to your knowledge?  Was thinking of knocking it out.

**Comment by ryno75** [comment]
https://github.com/terraform-providers/terraform-provider-aws/pull/9250

**Comment by ewbankkit** [comment]
Additional lifecycle policies [announced](https://aws.amazon.com/about-aws/whats-new/2019/07/optimize-cost-amazon-efs-infrequent-access-lifecycle-management/).
Requires:
* https://github.com/terraform-providers/terraform-provider-aws/pull/9287

**Comment by tdmalone** [comment]
^ And the above announcement also releases IA to _existing_ file systems created before the previous 13th Feb cutoff. 🎉

**Comment by ckappen** [comment]
on 9th of jly it was another announcement... 4 stages of infrequent access are available now in every region with efs
* after 14 days
* after 30 days
* after 60 days
* and after 90 days

**Comment by ryno75** [comment]
@ewbankkit do you think it would be best to include the SDK version bump and new lifecycles as part of my PR (#9250)?

**Comment by ewbankkit** [comment]
@ryno75 Yes it looks like you should `git rebase` onto the upstream master to address some of the changes that @bflad suggested.

**Comment by bflad** [comment]
Support for a new `lifecycle_policy` configuration block has been merged and will release with version 2.23.0 of the Terraform AWS Provider, in the next day or two. Thanks to @ryno75 for the implementation. 👍
